cpp/arcticdb/storage/storages.hpp

```diff
@@ -280,9 +280,9 @@ inline std::shared_ptr<Storages> create_storages(const LibraryPath& library_path
                                                  decltype(std::declval<arcticc::pb2::storage_pb2::LibraryConfig>().storage_by_id())& storage_configs,
                                                  const NativeVariantStorage& native_storage_config) {
     Storages::StorageVector storages;
-    for (const auto& [storage_id, storage_config] : storage_configs) {
+    for (auto& [storage_id, storage_config] : storage_configs) {
         util::variant_match(native_storage_config.variant(),
-                            [&storage_config, &storages, &library_path, mode](const s3::S3Settings& settings) {
+                            [storage_config, &storages, &library_path, mode](const s3::S3Settings& settings) {
```

Comment:
I notice we've changed to capturing the `storage_config` by value, but I think that's fine here - this is nowhere near a critical section of the code.